src/common/createplay/CreatePlay.jsx

```diff
@@ -1,56 +1,106 @@
-import { useState } from "react";
+// PACKAGES
+import { useReducer, useEffect } from "react";
 import { useAuthenticationStatus, useUserData } from "@nhost/react";
-import { Tags, Levels, Issues } from "common/services";
-import Button from "@mui/material/Button";
+import { useNavigate, useLocation, useParams } from "react-router-dom";
 
+// COMPONENTS & FILES
 import PlayForm from "common/components/PlayForms";
-import { NHOST } from "common/const";
-import { FIELD_TEMPLATE } from "./create-play-form-template";
-import "./create-play.scss";
 import Loader from "common/spinner/spinner";
-import { Plays } from "common/services/plays";
-import { useNavigate } from "react-router-dom";
 import { ReactComponent as NotAllowedImage } from "../../images/img-403.svg";
+import PageNotFound from "common/404/PageNotFound";
+import { Button } from "@mui/material";
+
+// UTILS
+import { FIELD_TEMPLATE } from "./create-play-form-template";
+import { defaultInputFields, createStateObject } from "./utils";
+
+// CONSTANTS
+import { NHOST } from "common/const";
+
+// SREVICES
+import { FetchPlaysBySlugAndUser } from "common/services/request/query/fetch-plays";
+import { Tags, Levels, Issues } from "common/services";
+import { Plays } from "common/services/plays";
+import { submit } from "common/services/request";
+import { updatePlayInfo } from "common/services/request/query/edit-play";
+
+// STYLES
+import "./create-play.scss";
```

Comment:
I very much liked these segregation and comments. Thanks

Reply:
ah thanks